pkg/controllers/controller.go

```diff
@@ -47,6 +47,13 @@ type Controller interface {
 	Owns() []Object
 }
 
+// NamedController allows distinguishing multiple controllers owning the same object
```

Comment:
NamedController allows controllers to optionally implement a Name() function which will be used instead of the reconciled resource's name. This is useful when writing multiple controllers for a single resource type.